libvast/vast/query.hpp

```diff
@@ -0,0 +1,99 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2021 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include "vast/expression.hpp"
+#include "vast/system/actors.hpp"
+
+#include <caf/typed_actor_view.hpp>
+
+namespace vast {
+
+/// A wrapper for an expression related command.
+struct query {
+  struct count {
+    enum mode { estimate, exact };
+    system::receiver_actor<uint64_t> sink;
+    enum mode mode = {};
+
+    friend bool operator==(const count& lhs, const count& rhs) {
+      return lhs.sink == rhs.sink && lhs.mode == rhs.mode;
+    }
+
+    template <class Inspector>
+    friend auto inspect(Inspector& f, count& x) {
+      return f(caf::meta::type_name("vast.query.count"), x.sink, x.mode);
+    }
+  };
+
+  struct extract {
+    enum ids_policy { drop, preserve };
```

Comment:
Does it make sense to name this one `mode` as well, renaming the values to `drop_ids` and `preserve_ids`?

Reply:
Makes sense.